torch/_inductor/codecache.py

```diff
@@ -249,35 +253,151 @@ def code_hash(code):
     )
 
 
-def get_code_path(source_code: Union[str, bytes], ext: str, extra: str, binary=False):
-    if binary:
-        source_code = repr(source_code)
-    basename = code_hash(source_code + extra)
+def serialize_fx_arg(arg: Any):
+    if isinstance(arg, torch.fx.GraphModule):
+        # Special case for fx_graph
+        return arg.print_readable(print_output=False).encode("utf-8")
+    if isinstance(arg, list):
+        if len(arg) > 0 and isinstance(arg[0], torch.Tensor):
+            # Special case for example_inputs, everything in torch/csrc/dynamo/guards.cpp
+            # Except for requires_grad, as that has already been handled
+            pickled_guards = []
+            for t in arg:
+                pickled_guards.append(
+                    b"".join(
+                        [
+                            pickle.dumps(torch._C._dispatch_keys(t).__repr__()),
+                            pickle.dumps(t.dtype),
+                            pickle.dumps(t.device),
+                            pickle.dumps(t.shape),
+                        ]
+                    )
+                )
+            return b"".join(pickled_guards)
+    return pickle.dumps(arg)
```

Comment:
I'm curious what debug mechanisms you have for when you are failing to cache hit, since you're using a pretty big hammer of pickle to serialize and it's easy for it to be nondeterministic.

Reply:
All the parameters that we have are either bools, ints, BoxedDeviceIndex, which translates to Optional[Int], and a frozenset. All of these should pickle deterministically, but I can make the serialize call a lot more explicit and make it fail through an assert when we add a new type and it isn't covered.